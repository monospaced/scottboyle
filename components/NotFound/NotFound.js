import React, {Component} from 'react';
import DocumentMeta from 'react-document-meta';

class NotFound extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    const {data} = this.props;
    return (
      <main className="Main">
        <DocumentMeta
          title={`Page not found | ${data.title} | ${data.subtitle}`}
          meta={{ name: { robots: 'noindex' } }}
        />
        <section>
          <h2>404 <small>Page not found</small></h2>
          <p>The requested URL was not found on this server.</p>
        </section>
      </main>
    );
  }
}

module.exports = NotFound;
