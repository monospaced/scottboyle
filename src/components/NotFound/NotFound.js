import DocumentMeta from "react-document-meta";
import React from "react";

import "../Main/Main.css";

const NotFound = ({ data: { subtitle, title } }) => {
  return (
    <main className="Main">
      <DocumentMeta
        title={`Page not found | ${title} | ${subtitle}`}
        meta={{ name: { robots: "noindex" } }}
      />
      <section>
        <h2>
          404 <small>Page not found</small>
        </h2>
        <p>The requested URL was not found on this server.</p>
      </section>
    </main>
  );
};

export default NotFound;
