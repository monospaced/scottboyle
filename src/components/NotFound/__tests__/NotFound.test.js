import React from "react";
import { shallow } from "enzyme";

import data from "../../../scripts/__mocks__/data.js";
import NotFound from "../NotFound.js";

describe("NotFound component", () => {
  const { subtitle, title } = data;

  it("should render correctly", () => {
    const props = { data: { subtitle, title } };
    const component = shallow(<NotFound {...props} />);
    expect(component).toMatchSnapshot();
  });
});
