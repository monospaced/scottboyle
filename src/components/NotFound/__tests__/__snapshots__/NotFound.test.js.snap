// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should render correctly 1`] = `
<main
  className="Main"
>
  <SideEffect(DocumentMeta)
    meta={
      Object {
        "name": Object {
          "robots": "noindex",
        },
      }
    }
    title="Page not found | Scott Boyle | Portfolio"
  />
  <section>
    <h2>
      404 
      <small>
        Page not found
      </small>
    </h2>
    <p>
      The requested URL was not found on this server.
    </p>
  </section>
</main>
`;
